src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -210,7 +211,24 @@ private T Deserialize<T>(JsonSerializer serializer, JToken jToken)
 
         private JsonSerializer GetPayloadSerializer(int? version)
         {
-            return version == 2 ? payloadSerializer2 : payloadSerializer;
+            if (version == null)
+            {
+                version = 1;
+            }
+
+            switch (version)
+            {
+                // 0 is used during negotiation
+                case 0:
+                case 1:
+                    return payloadSerializer;
+                case 2:
+                case 3:
+                    return payloadSerializer2;
+                default:
+                    throw new NotSupportedException($"Protocol version {version} is not supported. " +
```

Comment:
Should we localize the exception message ?

Reply:
Imho not necessary.